libs/partners/chroma/tests/integration_tests/test_vectorstores.py

```diff
@@ -414,3 +420,73 @@ def test_chroma_legacy_batching() -> None:
     )
 
     db.delete_collection()
+
+
+def test_create_collection_if_not_exist_default() -> None:
+    """Tests existing behaviour without the new create_collection_if_not_exists flag."""
+    texts = ["foo", "bar", "baz"]
+    docsearch = Chroma.from_texts(
+        collection_name="test_collection", texts=texts, embedding=FakeEmbeddings()
+    )
+    assert docsearch._client.get_collection("test_collection") is not None
+    docsearch.delete_collection()
+
+
+def test_create_collection_if_not_exist_true_existing(
+    client: chromadb.ClientAPI,
+) -> None:
+    """Tests create_collection_if_not_exists=True and collection already existing."""
+    client.create_collection("test_collection")
+    vectorstore = Chroma(
+        client=client,
+        collection_name="test_collection",
+        embedding_function=FakeEmbeddings(),
+        create_collection_if_not_exists=True,
+    )
+    assert vectorstore._client.get_collection("test_collection") is not None
+    vectorstore.delete_collection()
+
+
+def test_create_collection_if_not_exist_false_existing(
+    client: chromadb.ClientAPI,
+) -> None:
+    """Tests create_collection_if_not_exists=False and collection already existing."""
+    client.create_collection("test_collection")
+    vectorstore = Chroma(
+        client=client,
+        collection_name="test_collection",
+        embedding_function=FakeEmbeddings(),
+        create_collection_if_not_exists=False,
+    )
+    assert vectorstore._client.get_collection("test_collection") is not None
+    vectorstore.delete_collection()
+
+
+def test_create_collection_if_not_exist_false_non_existing(
+    client: chromadb.ClientAPI,
+) -> None:
+    """Tests create_collection_if_not_exists=False and collection not-existing,
+    should raise."""
+    with pytest.raises(Exception, match="does not exist"):
+        Chroma(
+            client=client,
+            collection_name="test_collection",
+            embedding_function=FakeEmbeddings(),
+            create_collection_if_not_exists=False,
+        )
+
+
+def test_create_collection_if_not_exist_true_non_existing(
+    client: chromadb.ClientAPI,
+) -> None:
+    """Tests create_collection_if_not_exists=True and collection non-existing. ."""
```

Comment:
Is this test correct? we have `create_collection_if_not_exists=False` below, and we create the collection.

Reply:
I think so, yes. if `create_collection_if_not_exists=False`, we'll do `get_collection` so the collection must exist; otherwise, Chroma will throw an error.

Reply:
This is easier to implement with hypothesis, but perhaps on the next iteration.

Reply:
Isn't this test identical to `test_create_collection_if_not_exist_false_existing`?

Sorry if I'm missing something.

Reply:
Nope, you're not missing anything. You are right that the test is identical even though its intended test scope (docstring) differs. Let me fix that.